backend/src/impl/benchmark_configs/config_gaokao.json

```
@@ -1,27 +1,640 @@
+    {
+      "dataset_name": "gaokao2021_npb",
+      "sub_dataset": "writing-grammar",
+      "dataset_split": "test",
+      "metrics": [{"name":"SeqCorrectCount"}],
+      "task": "grammatical-error-correction",
+      "output_file_type": "json"
+    },
+    {
+      "dataset_name": "gaokao2021_npb",
+      "sub_dataset": "writing-essay",
+      "dataset_split": "test",
+      "metrics": [{"name":"bleu"}],
+      "task": "conditional-generation",
+      "output_file_type": "text"
+    }
+  ],
+  "views": [
+      {
+          "name": "Overall with Essay",
+          "operations": [
+              {"op": "weighted_sum", "group_by": ["dataset_name"], "weight": "sub_dataset", "weight_map": {
+                  "listening":  1.5,
+                   "cloze-multiple-choice": 1.5,
+                   "cloze-hint":1.5,
+                   "reading-multiple-choice":2,
+                   "reading-cloze":2,
+                   "writing-grammar":1,
+                   "writing-essay":1
+                   }},
                 {"op": "mean"}
-            ]
-        },
-        {
-            "name": "Task-wise Average",
-            "operations": [
-                {"op": "mean", "group_by": ["dataset_name", "task"]},
-                {"op": "mean", "group_by": "task"}
-            ]
-        }
-    ],
-    "default_views": ["Overall"]
+          ]
+      },
+      {
+        "name": "Overall without Essay",
+        "operations": [
+          {"op": "weighted_sum", "group_by": ["dataset_name"], "weight": "sub_dataset", "weight_map": {
+              "listening":  1.5,
+               "cloze-multiple-choice": 1.5,
+               "cloze-hint":1.5,
+               "reading-multiple-choice":2,
+               "reading-cloze":2,
+               "writing-grammar":1,
+               "writing-essay":0
+               }},
+            {"op": "mean"}
+      ]
+    },
+      {
+          "name": "Question Type (with Essay)",
+          "operations": [
+              {"op": "mean", "group_by": "sub_dataset"},
+              {"op": "multiply", "weight": "sub_dataset", "weight_map": {
+                  "listening":  1.5,
+                  "cloze-multiple-choice": 1.5,
+                  "cloze-hint":1.5,
+                  "reading-multiple-choice":2,
+                  "reading-cloze":2,
+                  "writing-grammar":1,
+                   "writing-essay":1
+                   }}
+          ]
+      },
+      {
+        "name": "Question Type (without Essay)",
+        "operations": [
+            {"op": "mean", "group_by": "sub_dataset"},
+            {"op": "multiply", "weight": "sub_dataset", "weight_map": {
+                "listening":  1.5,
+                "cloze-multiple-choice": 1.5,
+                "cloze-hint":1.5,
+                "reading-multiple-choice":2,
+                "reading-cloze":2,
+                "writing-grammar":1,
+                 "writing-essay":0
+                 }}
+        ]
+    }
+  ],
+  "default_views": ["Overall"]
```

I know `default_views` has no effect now, but maybe we could future-proof as follows:

```suggestion
  "default_views": ["Overall with Essay", "Overall without Essay"]
```

---

agree and done!